### PR TITLE
[SOAR-8726] Carbon Black Defense Input Description Update

### PR DESCRIPTION
**Pull request by ablakley-r7**
### Description

Describe the proposed changes:

  - Update description of Time Range and Event IDs inputs

## PR Requirements

Developers, verify you have completed the following items by checking them off:

#### Unit Tests

Review our documentation on [generating](https://docs.rapid7.com/insightconnect/unit-test-generation) and [writing](https://docs.rapid7.com/insightconnect/unit-test-primer) plugin unit tests

- [x] Unit tests written for any new or updated code

#### In-Product Tests

If you are an InsightConnect customer or have access to an InsightConnect instance, the following in-product tests should be done:

- [x] Screenshot of job output with the plugin changes
- [x] Screenshot of the changed connection, actions, or triggers input within the InsightConnect workflow builder

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [x] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [x] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [x] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [x] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [x] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [x] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [x] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [x] Work fully completed
- [x] Functional
  - [x] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `icon-plugin run -c sample $action > tests/$action.json`
  - [x] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [x] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [x] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -T all --debug --jq` (use PR format at end)
  - [x] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -R all --debug --jq` (use PR format at end)

### Assessment

You must validate your work to reviewers:

1. Run `make validate` and make sure everything passes
2. Run the assessment tool: `icon-plugin run -A -R all -T all`. For single action validation: `icon-plugin run -A -R tests/my_action.json -T tests/my_action.json`
3. Copy (`icon-plugin ... | pbcopy`) and paste the output in **a new post** on this PR
4. Add required screenshots from the In-Product Tests section
Screenshot 2022-03-09 at 13 56 25
Screenshot 2022-03-09 at 13 48 59
Screenshot 2022-03-09 at 13 46 48

### CODE REVIEW

File: plugins/carbon_black_defense/help.md [hunk 1]
```diff
@@ -230,7 +230,7 @@ This action is used to retrieve details for an individual event given the event
 
 |Name|Type|Default|Required|Description|Enum|Example|
 |----|----|-------|--------|-----------|----|-------|
-|event_ids|string|None|True|Event ID|None|422af3fc3a7411ea8da649e797467dc0|
+|event_ids|string|None|True|Event ID used to retrieve event details. Example: 422af3fc3a7411ea8da649e797467dc0|None|422af3fc3a7411ea8da649e797467dc0|
```
Comment (cmcnally-r7) [inline plugins/carbon_black_defense/help.md:233]: Make sure example is only in example field and not in description field

File: plugins/carbon_black_defense/plugin.spec.yaml [hunk 2]
```diff
@@ -494,7 +494,7 @@ actions:
       time_range:
         title: "Time Range"
         type: string
-        description: "Time frame to limit your search to. Integer used for amount of of the following: s=second(s), m=minute(s), h=hour(s), d=day(s), w=week(s), or y=year(s)"
+        description: "Time frame to limit your search to. Integer used for amount of the following: s=second(s), m=minute(s), h=hour(s), d=day(s), w=week(s), or y=year(s). Example: -2d"
```
Comment (cmcnally-r7) [inline plugins/carbon_black_defense/plugin.spec.yaml:497]: Same as above, remove this example value as its already in `example`

File: plugins/carbon_black_defense/plugin.spec.yaml [hunk 3]
```diff
@@ -514,7 +514,7 @@ actions:
     input:
       event_ids:
         title: Event ID
-        description: Event ID
+        description: "Event ID used to retrieve event details. Example: 422af3fc3a7411ea8da649e797467dc0"
```
Comment (cmcnally-r7) [inline plugins/carbon_black_defense/plugin.spec.yaml:517]: Same again

**Review by cmcnally-r7 — CHANGES_REQUESTED** [review]
Remove examples from description fields

File: plugins/carbon_black_defense/help.md (hunk 1 above)
Comment (ablakley-r7) [reply plugins/carbon_black_defense/help.md:233]: Updated to include in description field as part of ticket request.

**Review by cmcnally-r7 — APPROVED** [review]
**Descriptions should include examples according to the ticket**
Everything minus this looks good 👍

File: plugins/carbon_black_defense/help.md [hunk 4]
```diff
@@ -15,7 +15,7 @@ This plugin utilizes the [VMware Carbon Black Cloud Endpoint Standard API](https
 
 # Supported Product Versions
 
-* 08/03/2022
+* 09/03/2022
```
Comment (blaxminarayan-r7) [inline plugins/carbon_black_defense/help.md:18]: this needs to be a timestamp - ISO format

File: plugins/carbon_black_defense/plugin.spec.yaml [hunk 5]
```diff
@@ -7,7 +7,7 @@ description: Find events and retrieve details for specific events
 version: 3.0.0
 vendor: rapid7
 support: rapid7
-supported_versions: ["08/03/2022"]
+supported_versions: ["09/03/2022"]
```
Comment (blaxminarayan-r7) [inline plugins/carbon_black_defense/plugin.spec.yaml:10]: Same as above

**Review by blaxminarayan-r7 — CHANGES_REQUESTED** [review]

**Review by blaxminarayan-r7 — APPROVED** [review]

**Pull request merged**